src/fetchPost.js

```diff
@@ -0,0 +1,365 @@
+import { deepSearchByKey, deepSearchByPair } from "./utils.js";
+let url = window.location.href;
+if (url.includes("post.html")) {
+  let url_split = url.split("=");
+  let postID = url_split[url_split.length - 1];
+
+  fetch(`http://localhost:9000/api/v8/posts-v2/web/${postID}`, {
+    headers: {
+      accept: "application/json-filled",
+      "accept-language": "en-US,en;q=0.9,fa;q=0.8",
+      "sec-ch-ua":
+        '" Not;A Brand";v="99", "Microsoft Edge";v="103", "Chromium";v="103"',
+      "sec-ch-ua-mobile": "?0",
+      "sec-ch-ua-platform": '"Linux"',
+      "sec-fetch-dest": "empty",
+      "sec-fetch-mode": "cors",
+      "sec-fetch-site": "same-site",
+    },
+    referrer: "https://divar.ir/",
+    referrerPolicy: "origin",
+    body: null,
+    method: "GET",
+    mode: "cors",
+    credentials: "omit",
+  })
+    .then((r) => r.json())
+    .then((b) => {
+      console.log(b);
+      let bookmarks = JSON.parse(window.localStorage.getItem("bookmarks"));
+      let notes = JSON.parse(window.localStorage.getItem("notes"));
+
+      let bookmarked = false;
+      if (bookmarks) {
+        for (let i = 0; i < bookmarks.length; i++) {
+          if (bookmarks[i].url == url) {
+            bookmarked = true;
+          }
+        }
+      }
+
+      let data = b.sections;
+      let title = deepSearchByKey(data, "title")["title"];
+      let author = deepSearchByKey(data, "subtitle")["subtitle"];
+      let chips = deepSearchByKey(data, "chips")["chips"];
+      let breadcrumb = deepSearchByPair(data, "widget_type", "BREADCRUMB")[0][
+        "data"
+      ]["parent_items"];
+      let infobox = null;
+      try {
+        infobox = deepSearchByPair(data, "widget_type", "GROUP_INFO_ROW")[0][
+          "data"
+        ]["items"];
+      } catch (error) {}
+
+      let rows = deepSearchByPair(data, "widget_type", "UNEXPANDABLE_ROW");
+      let description = deepSearchByPair(
+        data,
+        "widget_type",
+        "DESCRIPTION_ROW"
+      )[0]["data"]["text"];
+      description = description.replaceAll("\n", "<br/>");
+
+      let hasImage =
+        deepSearchByPair(data, "widget_type", "IMAGE_CAROUSEL").length > 0;
+      let images = hasImage
+        ? deepSearchByPair(data, "widget_type", "IMAGE_CAROUSEL")[0]["data"][
+            "items"
+          ]
+        : [];
+
+      let container = document.querySelector(".single-post-container");
+      container.innerHTML = `
+      <main class="single-post">
+        <nav class="single-post__breadcrumb">
+          ${breadcrumb
+            .map((item) => {
+              return `
+          <span class="single-post__arrowed-text"> ${
```

Comment:
The `arrowed-text` doesn't belong to `single-post`, but belongs to "breadcrumb".  
Breadcrumb should be a block of its own. It's independent and if we were using React it could've been a separate component. `single-post__breadcrumb` defines where the breadcrumb would be placed, but the `breadcrumb` block defines how it will look like.  
The same goes for other blocks under single-post as well, you didn't recognize them as independent self-containing blocks.